src/client/cli/cmd/mount.cpp

```diff
@@ -58,14 +58,17 @@ mp::ReturnCode cmd::Mount::run(mp::ArgParser* parser)
 
     mp::AnimatedSpinner spinner{cout};
 
-    auto on_success = [&spinner](mp::MountReply& reply) {
+    auto on_success = [this, &spinner](mp::MountReply& reply) {
         spinner.stop();
+        this->request.clear_target_paths();
+        this->request.clear_mount_maps();
```

Comment:
This is actually a weird situation. In the [run_cmd](https://github.com/canonical/multipass/blob/00c779694579942e95d0af6455035c8be90cdccb/src/client/cli/cmd/common_cli.cpp#L107-L117) function, we create a new parser, but we pass to it the same list of command objects from the main parser. So each time one of the `--mount` options runs the `mount` command, it will use the same command object, thus the same `MountRequest` object. This means that each new `--mount` will not only try to mount its own path, but the previous ones as well, which would result in an error. That is why after each execution of the command, I clear the targets from the request. A better alternative, in my opinion, would be to make sure that the new parser in this function is constructed with a fresh set of commands instead of getting references to the ones from the main parser. But I am not sure if this would cause problems somewhere else.